### PR TITLE
Use relative paths when generating autoloader

### LINKED ISSUE #3: Allow nested Composer dependencies that use composer-wordpress-autoloader
Currently, only the root level can be registered to use the autoloader.

### PR DESCRIPTION
**Pull request by srtfisher**
- [x] Use a relative path in favor of a hard-coded one to allow easy deployments.
- [x] Add support for autoloading inside of the `extra` part of your `composer.json`
- [x] Add support for nested autoloaders that don't use a full path. Fixes #3

Most of this is a port of how composer/composer works underneath.

### CODE REVIEW

File: src/AutoloadGenerator.php [hunk 1]
```diff
@@ -2,30 +2,268 @@
...
+     * @param boolean $isDevMode Flag if dev dependencies are being included.
+     * @return string
+     */
+    public function generate(bool $beingInjected, bool $isDevMode = true): string
+    {
+        $filesystem = new Filesystem();
+        $autoloadFileContents = '';
+
+        $basePath = $filesystem->normalizePath(realpath(realpath(getcwd())));
+        $vendorPath = $filesystem->normalizePath(realpath(realpath($this->composer->getConfig()->get('vendor-dir'))));
```
Comment (mogmarsh) [inline src/AutoloadGenerator.php:57]: I assume calling `realpath` twice on each is intentional?

File: src/AutoloadGenerator.php [hunk 2]
```diff
@@ -2,30 +2,268 @@
...
+));
+
+return \$autoload;
+AUTOLOAD;
+
+        return $autoloadFileContents . "\n";
+    }
+
+    /**
+     * Collect the autoloader rules to generator for.
```
Comment (mogmarsh) [inline src/AutoloadGenerator.php]: this comment doesn't read right to me.

File: src/Plugin.php [hunk 3]
```diff
@@ -129,61 +132,83 @@ protected function getAutoloaderFilePath(): string
...
-
-    /**
-     * Generate the autoloader file given a set of rules.
-     *
-     * @param array<string, string> $rules Autoloader rules.
-     * @param string $beingInjected Flag if the autoloader is being injected.
-     * @return string
-     */
-    protected function getAutoloaderFileContents(array $rules, bool $beingInjected): string
+    // /**
```
Comment (mogmarsh) [inline src/Plugin.php]: lots of commented out code. Should this be removed?

**Review by mogmarsh — APPROVED** [review]
a couple little things, plus a `.DS_Store` file to remove from the repo. Otherwise 👍

File: src/AutoloadGenerator.php (hunk 1 above)
Comment (srtfisher) [reply src/AutoloadGenerator.php:57]: It is. It's copy+paste from the composer code that did the same

**Pull request merged**